ocaml/xcp-rrdd/lib/transport/base/rrd_protocol_v2.ml

```diff
@@ -75,7 +75,8 @@ module Read = struct
   let datasource_count cs =
     Int32.to_int (Cstruct.BE.get_uint32 cs datasource_count_start)
 
-  let timestamp cs = Cstruct.BE.get_uint64 cs timestamp_start
+  let timestamp cs =
+    Int64.float_of_bits (Cstruct.BE.get_uint64 cs timestamp_start)
```

Comment:
Thinking about this... I don't think we can get away with the breaking change, there are python-based (GFS2) and C-based plugins (not sure which ones they are) :(

Reply:
After a discussion with the storage team, the timestamp-responsible part is in `rrdd.py` that we control, we'd just need to get a script to restart their metrics collection after update, so this should be do-able